server/methods.js

```diff
@@ -215,38 +215,69 @@ Meteor.methods({
 
   */
   flickrImageSearchList: function(query, option, page) {
-    var items, nextPage;
+    var items, nextPage, keywordSearch, path, requestParams;
     check(query, String);
+
+    if ((query.indexOf('flickr.com') !==-1) && (query.indexOf('/photos/') !==-1)){
+      keywordSearch = false;
+      //search photo: flickr.com/photos/{user-id}/{photo-id}/in/photolist-{search-info}
+      //individual photo:  flickr.com/photos/{user-id}/{photo-id}
+      var split = _.chain(query.split('/')).compact().value();
+      var offset = split.indexOf('photos');
+      if (split[offset+2]) var photo_id = (split[offset+2]).match(/[\d]*/)[0];
```

Comment:
can this use wrapping {} for clarity? also, what happens if the conditional is false?